packages/components/src/FormField/FormField.tsx

```diff
@@ -1,426 +1,216 @@
-   */
-  readonly value?: string | number;
-
-  /**
-   * Show a spinner to indicate loading
-   */
-  loading?: boolean;
-
-  /**
-   * Simplified onChange handler that only provides the new value.
-   * @param newValue
-   */
-  onChange?(newValue: string | number): void;
-
-  /**
-   * A callback to handle "Enter" keypress. This will only run
-   * if Enter is the only key. Will not run if Shift or Control
-   * are being held.
-   */
-  onEnter?(event: React.KeyboardEvent): void;
-
-  /**
-   * Focus callback.
-   */
-  onFocus?(): void;
-
-  /**
-   * Blur callback.
-   */
-  onBlur?(): void;
-
-  /**
-   * Callback to get the the status and message when validating a field
-   * @param message
-   */
-  onValidation?(message: string): void;
-
-  /**
-   * Show an error message above the field. This also
-   * highlights the the field red if an error message shows up.
-   */
-  readonly validations?: ValidationRules;
-
-  inputRef?: RefObject<
-    HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
-  >;
-}
-
-/**
- * Max statements is disabled for this component due the event handlers.
- */
-// eslint-disable-next-line max-statements
-export function FormField({
-  align,
-  autocomplete = true,
-  children,
-  defaultValue,
-  disabled,
-  inline,
-  invalid,
-  max,
-  maxLength,
-  min,
-  name,
-  onFocus,
-  onBlur,
-  onChange,
-  onEnter,
-  onValidation,
-  loading,
-  placeholder,
-  readonly,
-  rows,
-  size,
-  type = "text",
-  keyboard,
-  value,
-  validations,
-  inputRef,
-}: FormFieldProps) {
+export function FormField(props: FormFieldProps) {
   const {
-    register,
-    setValue,
-    control,
-    /**
-     * We currently only use `errors` from formState, but there are lots of
-     * other values that come with it. https://react-hook-form.com/api#formState
-     */
-    formState: { errors },
-  } =
+    actionsRef,
+    autocomplete = true,
+    children,
+    defaultValue,
+    disabled,
+    inline,
+    inputRef,
+    keyboard,
+    loading,
+    max,
+    maxLength,
+    min,
+    name,
+    readonly,
+    rows,
+    type = "text",
+    validations,
+    value,
+    onChange,
+    onEnter,
+    onFocus,
+    onBlur,
+    onValidation,
+  } = props;
+
+  const { control, errors, setValue } =
     useFormContext() != undefined
       ? useFormContext()
       : useForm({ mode: "onTouched" });
 
-  const [hasMiniLabel, setHasMiniLabel] = useState(
-    shouldShowMiniLabel(defaultValue, value),
-  );
   const [identifier] = useState(uuid.v1());
-  const isControlled = value !== undefined;
 
-  if (!name && (validations || isControlled)) {
-    name = `generatedName--${identifier}`;
-  }
-
-  const error = name && errors[name] && errors[name].message;
+  /**
+   * Generate a name if one is not supplied, this is the name
+   * that will be used for react-hook-form and not neccessarily
+   * attached to the DOM
+   */
+  const [controlledName] = useState(
+    name ? name : `generatedName--${identifier}`,
+  );
```

Comment:
Can we move this to a hook? Here is how we have done it on the mobile side:
https://github.com/GetJobber/jobber-mobile/blob/c0f73a4957d898a3c2671b9d93faf158320b90e7/src/hooks/useFormController/useFormController.ts

Reply:
@hosseintalebi we do require this `identifier` to be the same in the `FormLabel` and the input. do you think that it would still be better to do something like:

```ts
const [identifier] = useState(uuid.v1());
const [controlledName] = useControlledName(name, identifier);

// hook

function useControlledName(name: string | undefined, identifier: string) {
   return name ? name : `generatedName--${identifier}`
}
```